api/tasks.go

```diff
@@ -750,6 +750,11 @@ func (tmpl *Template) Canonicalize() {
 	if tmpl.Envvars == nil {
 		tmpl.Envvars = boolToPtr(false)
 	}
+
+	//COMPAT(0.12) VaultGrace is deprecated and unused as of Vault 0.5
+	if tmpl.VaultGrace == nil {
+		tmpl.VaultGrace = timeToPtr(0)
+	}
```

Comment:
This removes the implicit parsing (sets VaultGrace to 0 whether it is set or unset) that the original contributer had added. Given the error message in structs.go, this seems correct just checking.

Reply:
Great catch! This is intentional. We need to canonicalize it here because jobspec parsing is skipped when submitting JSON jobs directly to the api.